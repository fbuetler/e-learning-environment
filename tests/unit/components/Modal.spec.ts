import { shallowMount } from "@vue/test-utils";
import Modal from "@/components/shared/Modal.vue";

describe("Modal.vue", () => {
  const headerSlot = "<div>Header</div>";
  const bodySlot = "<div>Body</div>";
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(Modal, {
      scopedSlots: {
        header: headerSlot,
        body: bodySlot,
      },
    });
  });

  it("is a Vue instance", () => {
    expect(wrapper.vm).toBeTruthy();
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("has correct content", () => {
    expect(wrapper.find("div.modal-header").html()).toContain(headerSlot);
    expect(wrapper.find("div.modal-body").html()).toContain(bodySlot);
  });

  it("handles mousedown correctly on modal-mask", async () => {
    await wrapper.find("div.modal-mask").trigger("mousedown");
    expect(wrapper.emitted("close")).toBeTruthy();
    expect(wrapper.emitted("close").length).toBe(1);
  });

  it("handles mousedown correctly on modal-wrapper", async () => {
    await wrapper.find("div.modal-wrapper").trigger("mousedown");
    expect(wrapper.emitted("close")).toBeTruthy();
    expect(wrapper.emitted("close").length).toBe(1);
  });

  it("handles mousedown correctly on modal-container", async () => {
    await wrapper.find("div.modal-container").trigger("mousedown");
    expect(wrapper.emitted("close")).toBeFalsy();
  });
});
