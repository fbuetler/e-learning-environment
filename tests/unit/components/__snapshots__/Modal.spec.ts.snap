// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal.vue renders correctly 1`] = `
<transition-stub
  name="modal"
>
  <div
    class="modal-mask"
  >
    <div
      class="modal-wrapper"
    >
      <div
        class="modal-container"
      >
        <div
          class="modal-header"
        >
          <div>
            Header
          </div>
        </div>
         
        <div
          class="modal-body"
        >
          <div>
            Body
          </div>
        </div>
      </div>
    </div>
  </div>
</transition-stub>
`;
